### PR TITLE
enhancment #501 refactoring teacher controller and service

### LINKED ISSUE #501: Refactor Teacher entity
Your task will be to refactor all files, related to **Teacher** entity, such as controller, service, responses, DTOs,  etc.

Your steps can be as following:
1. Check service and repository class for potential issues, misspeling or incorect code style.
2. You need to run unit tests for this service and try to enhance them, also check that all necessary service methods are covered.
3. Check responses and DTOs. As we have a lot of changes some methods can return and take undocumentated information.
4. After all responses and DTOs are checked you can separate documentation of endpoints in controller as made in [this](https://github.com/fictadvisor/fictadvisor/pull/497) pr.

### PR DESCRIPTION
**Pull request by SylvaticGhost**
- refactored teacher service and test for it

closes #501

### CODE REVIEW

**Review by bohdanbulakh — CHANGES_REQUESTED** [review]
refactor controller, dtos, responses and swagger documentation, [here](https://github.com/fictadvisor/fictadvisor/pull/524) is an example

**Comment by SylvaticGhost** [comment]
@bohdanbulakh, I haven't realised the difference between controller and docs in example PR with the current state here. All required changes of the controller was provided in PR #460

File: fictadvisor-api/src/v2/api/services/TeacherService.spec.ts
Comment (bohdanbulakh) [inline fictadvisor-api/src/v2/api/services/TeacherService.spec.ts:1]: line 249: change description to smth like `should not throw an error ...`

File: fictadvisor-api/src/v2/mappers/TeacherMapper.ts
Comment (bohdanbulakh) [inline fictadvisor-api/src/v2/mappers/TeacherMapper.ts:1]: line 67: add `CathedraResponse[]` as  `getCathedras` method return type

File: fictadvisor-api/src/v2/api/services/TeacherService.spec.ts
Comment (bohdanbulakh) [inline fictadvisor-api/src/v2/api/services/TeacherService.spec.ts:1]: add tests for `getTeacherSubject`, `getAllContacts`, `getUserDisciplineTeachers`, `updateRating` methods

**Review by bohdanbulakh — CHANGES_REQUESTED** [review]

**Comment by SylvaticGhost** [comment]
@bohdanbulakh

**Review by bohdanbulakh — APPROVED** [review]
format `TeacherService.spec` and `TeacherService` with eslint

**Pull request merged**